substrate/frame/contracts/uapi/Cargo.toml

```diff
@@ -0,0 +1,23 @@
+[package]
+name = "pallet-contracts-uapi"
+version = "4.0.0-dev"
+authors.workspace = true
+edition.workspace = true
+license = "Apache-2.0"
+homepage = "https://substrate.io"
+repository.workspace = true
+description = "Exposes all the host functions that a contract can import."
+
+[dependencies]
+scale-info = { version = "2.10.0", default-features = false, features = ["derive"], optional = true  }
+scale = { package = "parity-scale-codec", version = "3.6.1", default-features = false, features = [
+	"derive",
+	"max-encoded-len",
+], optional = true }
+paste = { version = "1.0", default-features = false }
+bitflags = "1.0"
+
+[features]
+default = [ "scale" ]
+scale = [ "dep:scale", "scale-info" ]
```

Comment:
I don't really see where this feature is used? Scale is just used unconditionally to encode parameters when necessary. This is only the case in RISC-V which will be added in a follow up.

But even then it should be an architecture dependent dependency and not a feature.

Reply:
yep my bad this is for RISC-V and should be architecture dependent